mnist.py

```diff
@@ -151,6 +160,49 @@ def _download(data_dir: str):
     asyncio.run(_download_dataset(data_dir))
 
 
-def load(data_dir: str) -> Tuple[np.ndarray]:
+def mnist_load(data_dir: str) -> Tuple[np.ndarray]:
     _download(data_dir)
     return _create_dataset(data_dir)
+
+
+def main():
+    logger.info("Fetching dataset")
+
+    x_train, y_train, x_test, y_test = mnist_load("data")
+
+    logger.info("Creating model")
+
+    model = NeuralNetwork(
+        layers=(
+            (Dense(128), ReLU()),
+            (Dense(128), ReLU()),
+            (Dense(1), Sigmoid()),
+        ),
+        loss=BinaryCrossEntropy(),
+        optimizer=Adam(learning_rate=0.3),
```

Comment:
And 0.3 might be too high for Adam, perhaps use a bit lower e.g. 0.001

Reply:
Also made change to that